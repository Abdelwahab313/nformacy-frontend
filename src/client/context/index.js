import React, { createContext, useContext, useReducer } from 'react';
import {
  SET_ERROR_MESSAGE,
  UPDATE_CLIENTS,
  UPDATE_CURRENT_CLIENT,
  VERIFY_CLIENT,
  VERIFY_CURRENT_CLIENT,
} from './actionTypes';

const ClientContext = createContext();
const defaultValues = {
  clients: [],
  errorMessage: '',
};

const ClientProvider = ({ children, initialValue }) => {
  const { Provider } = ClientContext;
  initialValue = { ...defaultValues, ...initialValue };
  return (
    <Provider value={useReducer(clientReducer, initialValue)}>
      {children}
    </Provider>
  );
};

const useClientState = () => {
  const context = useContext(ClientContext);
  if (context === undefined) {
    throw new Error('useClientState must be used within a ClientProvider');
  }
  return context;
};

const clientReducer = (state, action) => {
  switch (action.type) {
    case UPDATE_CLIENTS:
      return {
        ...state,
        clients: action.payload,
      };
    case UPDATE_CURRENT_CLIENT:
      return {
        ...state,
        currentClient: action.payload,
      };
    case VERIFY_CURRENT_CLIENT:
      const currentClient = state.currentClient;
      currentClient.verified = true;
      return {
        ...state,
        currentClient: currentClient,
      };
    case VERIFY_CLIENT:
      const newClients = state.clients.map((client) => {
        if (client[0] === action.payload.uuid) {
          client[6] = true;
          return client;
        } else {
          return client;
        }
      });
      return {
        ...state,
        clients: newClients,
      };
    case SET_ERROR_MESSAGE:
      return {
        ...state,
        errorMessage: action.payload,
      };
  }
};

export { ClientProvider, useClientState };
