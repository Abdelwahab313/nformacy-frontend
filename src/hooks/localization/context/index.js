import React, { createContext, useContext, useReducer } from 'react';

const LocaleContext = createContext();
const DEFAULT_LOCALE = 'en';
const LOCALES = {
  en: 'en',
  ar: 'ar',
};
export const CHANGE_LOCALE = 'CHANGE_LOCALE';

export const useLocaleContext = () => {
  const context = useContext(LocaleContext);
  if (context === undefined)
    throw new Error('useLocale should only be used within locale provider');
  return context;
};

export const LocaleProvider = ({
  children,
  initialLocale = DEFAULT_LOCALE,
}) => {
  const { Provider } = LocaleContext;
  return (
    <Provider value={useReducer(localeReducer, initialLocale)}>{children}</Provider>
  );
};

const localeReducer = (state, action) => {
  if (action.type === CHANGE_LOCALE) {
    return action.payload.locale;
  }
  return LOCALES.en;
};
