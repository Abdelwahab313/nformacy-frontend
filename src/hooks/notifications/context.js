import React, { useContext, useReducer } from 'react';
import { ActionCableProvider } from '../useActionCable';
import { CHANNEL_URL } from '../../settings';
import { produce, current } from 'immer';
import { toast as showToast } from 'react-toastify';

const defaultStates = {
  notifications: [],
  menuOpened: null,
  unread: false,
  unreadCount: 0,
};

const NotificationsContext = React.createContext();

export const notificationActions = {
  notificationReceived: 'notificationReceived',
  menuToggled: 'menuToggled',
  menuClosed: 'menuClosed',
};

const MAX_NOTIFICATIONS = 10;
export const useNotificationsContext = () => {
  const context = useContext(NotificationsContext);
  if (context === undefined) {
    throw new Error(
      'useNotificationsContext must be used within a NotificationsProvider',
    );
  }
  return context;
};
const initContext = (initialValue) => {
  return initialValue || defaultStates;
};

export const NotificationsProvider = ({
  children,
  initialNotifications = [],
  unreadCount = 0,
}) => {
  const { Provider } = NotificationsContext;
  const initialState = {
    ...defaultStates,
    notifications: initialNotifications,
    unread: unreadCount > 0,
    unreadCount,
  };
  return (
    <Provider
      value={useReducer(NotificationsReducer, initialState, initContext)}>
      <ActionCableProvider url={CHANNEL_URL}>{children}</ActionCableProvider>
    </Provider>
  );
};

function Notification(notification) {
  return {
    notificationId: notification.notification_id,
    targetId: notification.target_id,
    messageKey: notification.message_key,
    createdAt: notification.created_at,
    type: notification.type,
    readAt: null,
  };
}

const updateUserInLocalStorage = (currentUser, updatedFields) => {
  localStorage.setItem(
    'user',
    JSON.stringify({ ...currentUser, ...updatedFields }),
  );
};

const NotificationsReducer = (state, action) => {
  switch (action.type) {
    case notificationActions.notificationReceived:
      const currentUser = action.payload.currentUser;
      const receivedNotification = Notification(action.payload.notification);
      showToast(receivedNotification.messageKey, {
        toastId: receivedNotification.notificationId,
      });
      return produce(state, (draftState) => {
        if (draftState.notifications.length === MAX_NOTIFICATIONS) {
          draftState.notifications.pop();
        }
        draftState.notifications.unshift(receivedNotification);
        draftState.unread = true;
        draftState.unreadCount = state.unreadCount + 1;
        updateUserInLocalStorage(currentUser, {
          unreadNotifications: draftState.unreadCount,
          notifications: current(draftState.notifications),
        });
      });
    case notificationActions.menuToggled:
      return produce(state, (draftState) => {
        if (state.menuOpened?.contains(action.payload.target)) {
          draftState.menuOpened = null;
        } else {
          draftState.menuOpened = action.payload.currentTarget;
          draftState.unread = false;
        }
      });
    case notificationActions.menuClosed:
      return produce(state, (draftState) => {
        draftState.menuOpened = null;
      });
    default:
      throw Error(
        `Action: ${action.type} is not defined in notifications reducer`,
      );
  }
};
