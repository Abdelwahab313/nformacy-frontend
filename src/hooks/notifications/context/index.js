import React, { useContext, useReducer } from 'react';
import { ActionCableProvider } from '../useActionCable';
import { CHANNEL_URL } from '../../../settings';
import {
  clearToast,
  closeMenu,
  loadNotifications,
  receiveNotification,
  toggleMenu,
  visitNotification,
} from './actions';

const defaultStates = {
  notifications: [],
  menuOpened: null,
  unread: false,
  unreadCount: 0,
};

const NotificationsContext = React.createContext();

export const notificationActions = {
  notificationsLoaded: 'notificationsLoaded',
  notificationReceived: 'notificationReceived',
  notificationVisited: 'notificationVisited',
  menuToggled: 'menuToggled',
  menuClosed: 'menuClosed',
  toastCleared: 'toastCleared',
};

export const useNotificationsContext = () => {
  const context = useContext(NotificationsContext);
  if (context === undefined) {
    throw new Error(
      'useNotificationsContext must be used within a NotificationsProvider',
    );
  }
  return context;
};
const initContext = (initialValue) => {
  return initialValue || defaultStates;
};

export const NotificationsProvider = ({ children }) => {
  const { Provider } = NotificationsContext;
  const initialState = {
    ...defaultStates,
    notifications: [],
    unreadCount: 0,
  };
  return (
    <Provider
      value={useReducer(NotificationsReducer, initialState, initContext)}>
      <ActionCableProvider url={CHANNEL_URL}>{children}</ActionCableProvider>
    </Provider>
  );
};

const NotificationsReducer = (state, action) => {
  switch (action.type) {
    case notificationActions.notificationsLoaded:
      return loadNotifications(state, action);
    case notificationActions.notificationReceived:
      return receiveNotification(action, state);
    case notificationActions.menuToggled:
      return toggleMenu(state, action);
    case notificationActions.menuClosed:
      return closeMenu(state);
    case notificationActions.notificationVisited:
      return visitNotification(state, action);
    case notificationActions.toastCleared:
      return clearToast(state);
    default:
      throw Error(
        `Action: ${action.type} is not defined in notifications reducer`,
      );
  }
};
