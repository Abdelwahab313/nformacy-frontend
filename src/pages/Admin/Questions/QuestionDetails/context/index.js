import React, { createContext, useContext, useReducer } from 'react';
import QuestionActionTypes from './actionTypes';

const defaultValues = {
  questionDetails: '',
};

const QuestionContext = createContext();

const QuestionProvider = ({ children, initialValue = {} }) => {
  initialValue = { ...defaultValues, ...initialValue };
  const { Provider } = QuestionContext;
  return (
    <Provider value={useReducer(QuestionReducer, initialValue)}>
      {children}
    </Provider>
  );
};

const useQuestionContext = () => {
  const context = useContext(QuestionContext);
  if (context === undefined) {
    throw new Error('useQuestionContext must be used within a AuthProvider');
  }
  return context;
};

const QuestionReducer = (state, action) => {
  switch (action.type) {
    case QuestionActionTypes.UPDATE_QUESTION:
      const updatedQuestion = { ...state.questionDetails, ...action.payload };
      return { ...state, questionDetails: updatedQuestion };

    default:
      throw new Error(`Unhandled action type: ${action.type}`);
  }
};

export { useQuestionContext, QuestionProvider };
