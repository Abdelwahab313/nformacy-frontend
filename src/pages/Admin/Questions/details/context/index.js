import React, { createContext, useContext, useReducer } from 'react';

const defaultValues = {
  questionDetails: '',
};

const QuestionContext = createContext();

const QuestionProvider = ({ children, initialValue = {} }) => {
  initialValue = { ...defaultValues, ...initialValue };
  const { Provider } = QuestionContext;
  return (
    <Provider value={useReducer(QuestionReducer, initialValue)}>
      {children}
    </Provider>
  );
};

const useQuestionContext = () => {
  const context = useContext(QuestionContext);
  if (context === undefined) {
    throw new Error('useQuestionContext must be used within a AuthProvider');
  }
  return context;
};

export const QuestionActionTypes = {
  UPDATE_QUESTION: 'UPDATE_QUESTION',
};

const QuestionReducer = (state, action) => {
  switch (action.type) {
    case QuestionActionTypes.UPDATE_QUESTION:
      return { ...state, questionDetails: action.payload };

    default:
      throw new Error(`Unhandled action type: ${action.type}`);
  }
};

export { useQuestionContext as useAuth, QuestionProvider as AuthProvider };
