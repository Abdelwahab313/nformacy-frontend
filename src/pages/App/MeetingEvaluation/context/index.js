import React, { createContext, useContext, useReducer } from 'react';
// import CallEvaluationActionTypes from './actionTypes';

const defaultValues = {
  comment: '',
  ratingEvaluations: {}
};

const CallEvaluationContext = createContext();

const CallEvaluationProvider = ({ children, initialValue = {} }) => {
  initialValue = { ...defaultValues, ...initialValue };
  const { Provider } = CallEvaluationContext;
  return (
    <Provider value={useReducer(CallEvaluationReducer, initialValue)}>
      {children}
    </Provider>
  );
};

const useCallEvaluationContext = () => {
  const context = useContext(CallEvaluationContext);
  if (context === undefined) {
    throw new Error('useCallEvaluationContext must be used within a CallEvaluationContext');
  }
  return context;
};

const CallEvaluationReducer = (state, action) => {
  switch (action.type) {

    default:
      throw new Error(`Unhandled action type: ${action.type}`);
  }
};

export { useCallEvaluationContext, CallEvaluationProvider };
