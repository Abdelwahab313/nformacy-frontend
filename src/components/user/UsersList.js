import React, { useEffect, useState } from 'react';
import {
  AppBar,
  Button,
  Dialog,
  Grid,
  IconButton,
  makeStyles,
  Slide,
  Toolbar,
  Typography,
} from '@material-ui/core';
import MaterialTable from 'material-table';
import { fetchUsers } from '../../apis/usersApi';
import { table_localization } from '../../settings';
import CloseIcon from '@material-ui/icons/Close';
import AddUserForm from './addUserForm';
import '../../index.css';
import { useAuth } from '../../context/auth';
import ErrorDialog from '../errors/ErrorDialog';
import _ from 'lodash';
import CircularProgress from '@material-ui/core/CircularProgress';

const useStyles = makeStyles((theme) => ({
  root: {
    background: '#F5F5F5',
    padding: theme.spacing(3),
  },
  addButton: {
    marginBottom: theme.spacing(1),
  },
  toolBar: {
    position: 'relative',
    color: '#edf0f2',
    justifyContent: 'space-between',
    flexBasis: '100%',
  },
  addForm: {
    margin: 'auto',
  },
  tableContainer: {
    width: '85%',
    margin: theme.spacing(1),
  },
  progressContainer: {
    position: 'absolute',
    left: '50%',
    top: '50%',
    transform: 'translate(-50%, -50%)',
  },
}));

const Transition = React.forwardRef(function Transition(props, ref) {
  return <Slide direction='up' ref={ref} {...props} />;
});

export default function FullScreenDialog() {
  const classes = useStyles();
  const [users, setUsers] = useState([]);
  const [open, setOpen] = React.useState(false);
  const [errorMessage, setErrorMessage] = useState('');
  const [showError, setShowError] = useState(false);
  const { authTokens, setAuthTokens } = useAuth();
  const [usersLoading, setUsersLoading] = useState(false);

  const handleClickOpen = () => {
    setOpen(true);
  };

  const handleClose = (createdUser) => {
    setOpen(false);
    if (createdUser.type !== 'click') {
      let tempUsers = _.cloneDeep(users);
      tempUsers.push(createdUser);
      setUsers(tempUsers);
    }
  };

  function getUsers() {
    setUsersLoading(true);
    return fetchUsers(authTokens)
      .then((res) => {
        const fetchedUsers = res.data;
        fetchedUsers.sort(
          (a, b) => new Date(a.date_joined) - new Date(b.date_joined),
        );
        setUsers(fetchedUsers);
      })
      .catch((reason) => {
        if (reason.response.status === 401) {
          localStorage.removeItem('tokens');
          localStorage.removeItem('users');
          setAuthTokens();
        }
      })
      .catch((reason) => {
        setErrorMessage('حدث خطأ أثناء الاتصال بالخادم');
        setShowError(true);
      });
  }

  useEffect(() => {
    getUsers().finally(() => {
      setUsersLoading(false);
    });
  }, []);
  if (usersLoading) {
    return (
      <div className={classes.progressContainer}>
        <CircularProgress />
      </div>
    );
  } else {
    return (
      <div dir='rtl' className={classes.root}>
        {showError && (
          <ErrorDialog
            message={errorMessage}
            close={() => {
              setShowError(false);
              setErrorMessage();
            }}
          />
        )}
        <Grid className={classes.tableContainer}>
          <Button
            className={classes.addButton}
            variant='contained'
            id={'add-user-button'}
            onClick={handleClickOpen}
            color='primary'>
            اضافه موظف جديد
          </Button>
          <MaterialTable
            id={'usersList'}
            localization={table_localization('موظفين')}
            columns={[
              {
                title: 'أسم المستخدم',
                field: 'name',
                render: (user) => {
                  return `${user.first_name} ${user.last_name}`;
                },
              },
              { title: 'رقم الموبايل', field: 'phone_number' },
              { title: 'رقم البطاقه', field: 'national_id' },
              { title: 'اسم تسجيل الدخول', field: 'username' },
              {
                title: 'تاريخ الاضافه',
                field: 'created',
                type: 'date',
                render: (user) => {
                  const date = new Date(user.date_joined);
                  return `${date.toLocaleTimeString()} ${date.toLocaleDateString()}`;
                },
              },
            ]}
            data={users}
            title={'المستخدمين'}
          />
        </Grid>
        <Dialog
          fullScreen
          open={open}
          onClose={handleClose}
          TransitionComponent={Transition}
          id={'add-user-dialog'}>
          <AppBar>
            <Toolbar className={classes.toolBar}>
              <IconButton edge='start' onClick={handleClose} aria-label='close'>
                <CloseIcon />
              </IconButton>
              <Typography>اضافه موظف جديد</Typography>
            </Toolbar>
          </AppBar>
          <AddUserForm onClose={handleClose} />
        </Dialog>
      </div>
    );
  }
}
