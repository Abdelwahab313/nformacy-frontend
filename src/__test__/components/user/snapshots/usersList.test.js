import React from 'react';
import UsersList from '../../../../components/user/UsersList';
import * as AuthContextMoudle from '../../../../context/auth';
import { AuthContext } from '../../../../context/auth';
import { shallow } from 'enzyme';
import { MemoryRouter, Redirect } from 'react-router';

function mockAuthObject() {
  jest.spyOn(AuthContextMoudle, 'useAuth').mockImplementation(() => ({
    authTokens: '',
  }));
}

const testUsersList = [
  {
    first_name: 'first name',
    last_name: 'second name',
    phone_number: '01099234802',
    national_id: '29505050103981',
    username: 'omarmmali',
    date_joined: '05/05/1995',
  },
];

it('renders correctly', () => {
  mockAuthObject();

  const container = shallow(
    <AuthContext.Provider>
      <UsersList usersList={testUsersList} />
    </AuthContext.Provider>,
  );
  const tree = container.html();
  expect(tree).toMatchSnapshot();
});

it('Should route to detail view correctly', () => {
  mockAuthObject();

  const container = shallow(
    <AuthContext.Provider>
      <UsersList usersList={testUsersList} />
    </AuthContext.Provider>,
  );

  expect(container.props().userDetailsRoute).toEqual('/users/');
});
