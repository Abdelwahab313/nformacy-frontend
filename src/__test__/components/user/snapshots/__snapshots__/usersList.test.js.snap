// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `"<div dir=\\"rtl\\" class=\\"makeStyles-root-1\\"><div class=\\"MuiGrid-root makeStyles-tableContainer-5\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-addButton-2 MuiButton-containedPrimary\\" tabindex=\\"0\\" type=\\"button\\" id=\\"add-user-button\\"><span class=\\"MuiButton-label\\">اضافه موظف جديد</span></button><div class=\\"MuiPaper-root MuiPaper-elevation2 MuiPaper-rounded\\" style=\\"position:relative\\"><div class=\\"MuiToolbar-root MuiToolbar-regular MTableToolbar-root-172 MuiToolbar-gutters\\"><div class=\\"MTableToolbar-title-176\\"><h6 class=\\"MuiTypography-root MuiTypography-h6\\">المستخدمين</h6></div><div class=\\"MTableToolbar-spacer-174\\"></div><div class=\\"MuiFormControl-root MuiTextField-root MTableToolbar-searchField-177\\"><div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd\\"><div class=\\"MuiInputAdornment-root MuiInputAdornment-positionStart\\"><span class=\\"material-icons MuiIcon-root MuiIcon-fontSizeSmall\\" aria-hidden=\\"true\\" title=\\"بحث\\">search</span></div><input type=\\"text\\" aria-invalid=\\"false\\" placeholder=\\"بحث\\" value=\\"\\" class=\\"MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd\\"/><div class=\\"MuiInputAdornment-root MuiInputAdornment-positionEnd\\"><button class=\\"MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled\\" tabindex=\\"-1\\" type=\\"button\\" disabled=\\"\\"><span class=\\"MuiIconButton-label\\"><span class=\\"material-icons MuiIcon-root MuiIcon-fontSizeSmall\\" aria-hidden=\\"true\\">clear</span></span></button></div></div></div><div class=\\"MTableToolbar-actions-175\\"><div><div><span></span></div></div></div></div><div style=\\"overflow-x:auto\\"><div><div style=\\"overflow-y:auto\\"><table class=\\"MuiTable-root\\"><thead class=\\"MuiTableHead-root\\"><tr class=\\"MuiTableRow-root MuiTableRow-head\\"><th class=\\"MuiTableCell-root MuiTableCell-head MTableHeader-header-287 MuiTableCell-paddingCheckbox\\" scope=\\"col\\" style=\\"text-align:center\\"><span class=\\"MuiButtonBase-root MuiTableSortLabel-root Mui-disabled\\" tabindex=\\"-1\\" role=\\"button\\" aria-disabled=\\"true\\">عمليات<svg class=\\"MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionDesc\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z\\"></path></svg></span></th><th class=\\"MuiTableCell-root MuiTableCell-head MTableHeader-header-287 MuiTableCell-alignLeft\\" scope=\\"col\\"><span class=\\"MuiButtonBase-root MuiTableSortLabel-root\\" tabindex=\\"0\\" role=\\"button\\" aria-disabled=\\"false\\"><div data-react-beautiful-dnd-draggable=\\"0\\" tabindex=\\"0\\" data-react-beautiful-dnd-drag-handle=\\"0\\" aria-roledescription=\\"Draggable item. Press space bar to lift\\" draggable=\\"false\\">أسم المستخدم</div><span class=\\"material-icons MuiIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc\\" aria-hidden=\\"true\\">arrow_downward</span></span></th><th class=\\"MuiTableCell-root MuiTableCell-head MTableHeader-header-287 MuiTableCell-alignLeft\\" scope=\\"col\\"><span class=\\"MuiButtonBase-root MuiTableSortLabel-root\\" tabindex=\\"0\\" role=\\"button\\" aria-disabled=\\"false\\"><div data-react-beautiful-dnd-draggable=\\"0\\" tabindex=\\"0\\" data-react-beautiful-dnd-drag-handle=\\"0\\" aria-roledescription=\\"Draggable item. Press space bar to lift\\" draggable=\\"false\\">رقم الموبايل</div><span class=\\"material-icons MuiIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc\\" aria-hidden=\\"true\\">arrow_downward</span></span></th><th class=\\"MuiTableCell-root MuiTableCell-head MTableHeader-header-287 MuiTableCell-alignLeft\\" scope=\\"col\\"><span class=\\"MuiButtonBase-root MuiTableSortLabel-root\\" tabindex=\\"0\\" role=\\"button\\" aria-disabled=\\"false\\"><div data-react-beautiful-dnd-draggable=\\"0\\" tabindex=\\"0\\" data-react-beautiful-dnd-drag-handle=\\"0\\" aria-roledescription=\\"Draggable item. Press space bar to lift\\" draggable=\\"false\\">رقم البطاقه</div><span class=\\"material-icons MuiIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc\\" aria-hidden=\\"true\\">arrow_downward</span></span></th><th class=\\"MuiTableCell-root MuiTableCell-head MTableHeader-header-287 MuiTableCell-alignLeft\\" scope=\\"col\\"><span class=\\"MuiButtonBase-root MuiTableSortLabel-root\\" tabindex=\\"0\\" role=\\"button\\" aria-disabled=\\"false\\"><div data-react-beautiful-dnd-draggable=\\"0\\" tabindex=\\"0\\" data-react-beautiful-dnd-drag-handle=\\"0\\" aria-roledescription=\\"Draggable item. Press space bar to lift\\" draggable=\\"false\\">اسم تسجيل الدخول</div><span class=\\"material-icons MuiIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc\\" aria-hidden=\\"true\\">arrow_downward</span></span></th><th class=\\"MuiTableCell-root MuiTableCell-head MTableHeader-header-287 MuiTableCell-alignLeft\\" scope=\\"col\\"><span class=\\"MuiButtonBase-root MuiTableSortLabel-root\\" tabindex=\\"0\\" role=\\"button\\" aria-disabled=\\"false\\"><div data-react-beautiful-dnd-draggable=\\"0\\" tabindex=\\"0\\" data-react-beautiful-dnd-drag-handle=\\"0\\" aria-roledescription=\\"Draggable item. Press space bar to lift\\" draggable=\\"false\\">تاريخ الاضافه</div><span class=\\"material-icons MuiIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc\\" aria-hidden=\\"true\\">arrow_downward</span></span></th></tr></thead><tbody class=\\"MuiTableBody-root\\"><tr class=\\"MuiTableRow-root\\" index=\\"0\\" level=\\"0\\" path=\\"0\\" style=\\"transition:all ease 300ms\\"><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-paddingNone\\" style=\\"width:42px;padding:0px 5px\\"><div style=\\"display:flex\\"><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit\\" tabindex=\\"0\\" type=\\"button\\" title=\\"تفاصيل الموزع\\"><span class=\\"MuiIconButton-label\\"><span class=\\"material-icons MuiIcon-root MuiIcon-colorPrimary\\" aria-hidden=\\"true\\">help</span></span></button></div></td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft\\" style=\\"color:inherit\\">first name second name</td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft\\" value=\\"01099234802\\" style=\\"color:inherit\\">01099234802</td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft\\" value=\\"29505050103981\\" style=\\"color:inherit\\">29505050103981</td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft\\" value=\\"omarmmali\\" style=\\"color:inherit\\">omarmmali</td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft\\" style=\\"color:inherit\\">12:00:00 AM 5/5/1995</td></tr><tr class=\\"MuiTableRow-root\\" style=\\"height:49px\\"></tr><tr class=\\"MuiTableRow-root\\" style=\\"height:49px\\"></tr><tr class=\\"MuiTableRow-root\\" style=\\"height:49px\\"></tr><tr class=\\"MuiTableRow-root\\" style=\\"height:49px\\"></tr><tr class=\\"MuiTableRow-root\\" style=\\"height:1px\\"></tr></tbody></table></div></div></div><table class=\\"MuiTable-root\\"><tfoot class=\\"MuiTableFooter-root\\" style=\\"display:grid\\"><tr class=\\"MuiTableRow-root MuiTableRow-footer\\"><td class=\\"MuiTableCell-root MuiTableCell-footer MuiTablePagination-root Component-paginationRoot-141\\" colSpan=\\"3\\" style=\\"float:right;overflow-x:auto\\" icons=\\"[object Object]\\"><div class=\\"MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar Component-paginationToolbar-142 MuiToolbar-gutters\\"><div class=\\"MuiTablePagination-spacer\\"></div><p class=\\"MuiTypography-root MuiTablePagination-caption Component-paginationCaption-143 MuiTypography-body2 MuiTypography-colorInherit\\">Rows per page:</p><div class=\\"MuiInputBase-root MuiTablePagination-input MuiTablePagination-selectRoot Component-paginationSelectRoot-144\\"><div class=\\"MuiSelect-root MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInputBase-input MuiInputBase-inputSelect\\" tabindex=\\"0\\" role=\\"button\\" aria-labelledby=\\" \\" aria-haspopup=\\"listbox\\"><div style=\\"padding:0px 5px\\">5 موظفين </div></div><input type=\\"hidden\\" value=\\"5\\"/><svg class=\\"MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M7 10l5 5 5-5z\\"></path></svg></div><p class=\\"MuiTypography-root MuiTablePagination-caption Component-paginationCaption-143 MuiTypography-body2 MuiTypography-colorInherit\\">1-1 من 1</p><div class=\\"MTablePaginationInner-root-346\\"><span title=\\"الصفحه الاولى\\" class=\\"\\"><button class=\\"MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled\\" tabindex=\\"-1\\" type=\\"button\\" disabled=\\"\\" aria-label=\\"الصفحه الاولى\\"><span class=\\"MuiIconButton-label\\"><span class=\\"material-icons MuiIcon-root\\" aria-hidden=\\"true\\">first_page</span></span></button></span><span title=\\"الصفحه السابقه\\" class=\\"\\"><button class=\\"MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled\\" tabindex=\\"-1\\" type=\\"button\\" disabled=\\"\\" aria-label=\\"الصفحه السابقه\\"><span class=\\"MuiIconButton-label\\"><span class=\\"material-icons MuiIcon-root\\" aria-hidden=\\"true\\">chevron_left</span></span></button></span><span class=\\"MuiTypography-root MuiTypography-caption\\" style=\\"flex:1;text-align:center;align-self:center;flex-basis:inherit\\">1-1 من 1</span><span title=\\"الصفحه التاليه\\" class=\\"\\"><button class=\\"MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled\\" tabindex=\\"-1\\" type=\\"button\\" disabled=\\"\\" aria-label=\\"الصفحه التاليه\\"><span class=\\"MuiIconButton-label\\"><span class=\\"material-icons MuiIcon-root\\" aria-hidden=\\"true\\">chevron_right</span></span></button></span><span title=\\"الصفحه الأخيره\\" class=\\"\\"><button class=\\"MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled\\" tabindex=\\"-1\\" type=\\"button\\" disabled=\\"\\" aria-label=\\"الصفحه الأخيره\\"><span class=\\"MuiIconButton-label\\"><span class=\\"material-icons MuiIcon-root\\" aria-hidden=\\"true\\">last_page</span></span></button></span></div></div></td></tr></tfoot></table></div></div></div>"`;
